### PR TITLE
Chore: upgrade to Flutter 3.24.3 and XCode 16.0

### LINKED ISSUE #284: iOS SDK version 16 (XCode 14) is not supported anymore
We should build using `macos-14` (the new latest) instead of `macos-13`, and we should use XCode 15 instead of XCode 14. We might also have to update Flutter when our current Flutter version has no support for XCode 15. Apple does not accept our builds anymore due to this.

Take into account that issue #273 should be resolved when upgrading.

MacOS runner: https://github.com/privacybydesign/irmamobile/blob/012ea862b94446e7da9fc3bd43801fb74336f1ca/.github/workflows/delivery.yml#L13 (and many other occurances)

XCode version: https://github.com/privacybydesign/irmamobile/blob/012ea862b94446e7da9fc3bd43801fb74336f1ca/ci_scripts/select_xcode.sh#L4

### PR DESCRIPTION
**Pull request by ivard**
Fixes #284

TODOs:

- [x] Lottie animation in IntroductionAnimationWrapper does not render correctly using Impeller renderer (I haven't checked `NameChangedScreen` so it might be there too)
```
[ERROR:flutter/impeller/base/validation.cc(59)] Break on 'ImpellerValidationBreak' to inspect point of failure: Requested texture size (8296, 2732) exceeds maximum supported size of (8192, 8192)
[ERROR:flutter/impeller/base/validation.cc(59)] Break on 'ImpellerValidationBreak' to inspect point of failure: Could not create multisample color texture.
[ERROR:flutter/impeller/base/validation.cc(59)] Break on 'ImpellerValidationBreak' to inspect point of failure: Render target does not have color attachment at index 0.
[ERROR:flutter/impeller/base/validation.cc(59)] Break on 'ImpellerValidationBreak' to inspect point of failure: Render target does not have color attachment at index 0.
[ERROR:flutter/impeller/base/validation.cc(59)] Break on 'ImpellerValidationBreak' to inspect point of failure: Subpass render target is invalid.
```

- [x] Find solution or apply workaround for failing integration tests on iOS due to #273

### CODE REVIEW

**Comment by w-ensink** [comment]
I have run the integration tests both on my physical device (iPhone 13 Pro) as well as in a simulator and both succeed without any problem. I'm running iOS 18.1 and use Xcode 16.1. 

The Firebase CI still fails, even after upgrading to Xcode 16.1. Maybe it's caused by using an older iPhone and iOS combi (iPhone 8 and iOS 15)

As for the Lottie animation: on my physical device I don't have any trouble with the Impeller engine, but on the simulator I do get the same error.
I have tried using a different Lottie animation and that didn't cause any problems. I also tried scaling the animation down, which seems fix it too.

The errors only begin when the animation zooms in at the end, which leads me to believe it renders at a much higher resolution at that point (e.g. rendering the whole thing and then zooming in on the result or something).

**Comment by ivard** [comment]
I tested it again locally with my iPhone 8 and XCode 16.1 and the test still fails after ~6 minutes. You have to run it by doing `flutter build ios ./integration_test/test_all.dart --config-only` and then run the test target in XCode on your physical iPhone. Did you also run the test via XCode?

**Comment by ivard** [comment]
Concerning the lottie animation, weird that your physical iPhone does not show the error. Maybe it only occurs on device with smaller screen dimensions.

**Comment by ivard** [comment]
If I use 'Product > Run' in XCode then the test runs the full 15 minutes, but when I use 'Product > Test' then the tests stop after ~6 minutes on my physical iPhone 8.

**Comment by w-ensink** [comment]
Interesting.. I picked use `Xcode -> Test` and it worked just fine. Do you know what the difference between these two/three options is?

**Comment by w-ensink** [comment]
I removed `integration-test-ios` from the branch protection rules, because it would never complete due to GitHub Actions not really understanding the matrix. I added a final job that checks the results of the matrix and then passes or failed based on that. I added that job to the branch protection rules so all iOS tests still need to pass before merge.

**Review by DibranMulder — APPROVED** [review]
Looks good we are dropping iOS 11 support but has been unsupported for quite some time now. The latest release date is from July 9, 2018 according to https://en.wikipedia.org/wiki/IOS_version_history

File: ios/Flutter/AppFrameworkInfo.plist [hunk 1]
```diff
@@ -21,6 +21,6 @@
   <key>CFBundleVersion</key>
   <string>1.0</string>
   <key>MinimumOSVersion</key>
-  <string>11.0</string>
+  <string>12.0</string>
```
[the remark this answers is not in the document]
Comment (w-ensink) [reply ios/Flutter/AppFrameworkInfo.plist:24]: None of our users have lower than iOS 17.

**Review by w-ensink — APPROVED** [review]

**Comment by ivard** [comment]
> Looks good we are dropping iOS 11 support but has been unsupported for quite some time now. The latest release date is from July 9, 2018 according to https://en.wikipedia.org/wiki/IOS_version_history

iOS 11 wasn't supported by us anymore anyway, because gomobile dropped support for this quite some time ago.

image

**Pull request merged**